tools/codegen/api/unboxing.py

```diff
@@ -0,0 +1,201 @@
+from typing import List, Tuple
+
+from tools.codegen.api import cpp
+from tools.codegen.api.types import Binding, CType, CppSignatureGroup
+from tools.codegen.model import (
+    Argument,
+    NativeFunction,
+    Type,
+    BaseType,
+    OptionalType,
+    ListType,
+    BaseTy,
+)
+
+# This file generates the code for unboxing wrappers, i.e., the glue logic to unbox a boxed operator and convert the
+# ivalues from stack to correct arguments to the unboxed kernel, based on corresponding JIT schema. This codegen is
+# an alternative way to generate unboxing wrappers similar to the existing C++ metaprogramming approach but gets the
+# job done statically. These generated unboxing wrappers will be useful under the scenario where we need to register
+# a fixed set of operators known at compile time and thus can save some time in runtime initialization phase.
+#
+# Here's an example on how the codegen works:
+#
+# - Function Schema (source of truth)
+#
+#      aten::empty.names(int[] size, *, Dimname[]? names,
+#                        ScalarType? dtype=None, Layout? layout=None,
+#                        Device? device=None, bool? pin_memory=None,
+#                        MemoryFormat? memory_format=None) -> Tensor
+# - Argument Conversion
+#       Generates C++ code to convert an ivalue (from stack) to its underlying C++ type.
+#    - int[] size
+#        ```cpp
+#           const c10::List<c10::IValue> size_list_in = (std::move(peek(stack, 0, 7))).toList();
```

Comment:
Where's 0 and 7 from? How do we know the which locations from the stack are the list elements?

Reply:
`gen.py` or rather the whole codegen framework provided all these information in Python APIs. Fundamentally the source of truth is `native_functions.yaml`.